src/datahike/db.cljc

```diff
@@ -1120,12 +1124,11 @@
                 (and keep-history? indexing?) (update-in [:temporal-avet] #(di/-insert % datom :avet)))
         db))))
 
-
 (defn- with-temporal-datom [db ^Datom datom]
   (let [indexing? (indexing? db (.-a datom))
         schema? (ds/schema-attr? (.-a datom))
         current-datom ^Datom (first (-search db [(.-e datom) (.-a datom) (.-v datom)]))
-        history-datom ^Datom (first (search-temporal-indices db [(.-e datom) (.-a datom) (.-v datom)]))
+        history-datom ^Datom (first (search-temporal-indices db [(.-e datom) (.-a datom) (.-v datom) (.-tx datom)]))
```

Comment:
So happened before you made this change?

Reply:
Right, the tx id in the removed datom was not the tx id from the when the removal happened but the original insertion tx id.

Reply:
I see, thanks